File: 705.design-hashset/solution.kt
class MyHashSet() {

    private val table = Array<MutableList<Int>?>(997) { null }
    
    fun add(key: Int) {
        val h = key.hashCode() % table.size
        val l = table[h] ?: mutableListOf<Int>()
        if (key !in l) {
            l.add(key)
            table[h] = l
        }
    }

    fun remove(key: Int) {
        val h = key.hashCode() % table.size
        table[h]?.remove(key)        
    }

    fun contains(key: Int): Boolean {
        val h = key.hashCode() % table.size
        val l = table[h]
        return l?.contains(key) ?: false
    }

}

/**
 * Your MyHashSet object will be instantiated and called as such:
 * var obj = MyHashSet()
 * obj.add(key)
 * obj.remove(key)
 * var param_3 = obj.contains(key)
 */

File: 705.design-hashset/solution.go
const size = 997

type MyHashSet struct {
    buckets [][]int
}


func Constructor() MyHashSet {
    return MyHashSet{make([][]int, size)}
}


func (this *MyHashSet) Add(key int)  {
    h := hash(key)
    for _, k := range this.buckets[h] {
        if k == key {
            return
        }
    }
    this.buckets[h] = append(this.buckets[h], key)
}


func (this *MyHashSet) Remove(key int)  {
    h := hash(key)
    b := this.buckets[h]
    for i := range b {
        if b[i] == key {
            this.buckets[h] = append(b[:i], b[i + 1:]...)
            return
        }
    }
}


func (this *MyHashSet) Contains(key int) bool {
    h := hash(key)
    for _, k := range this.buckets[h] {
        if k == key {
            return true
        }
    }
    return false
}


func hash(key int) int {
    return key % size
}

/**
 * Your MyHashSet object will be instantiated and called as such:
 * obj := Constructor();
 * obj.Add(key);
 * obj.Remove(key);
 * param_3 := obj.Contains(key);
 */